api/client/commands.go

```diff
@@ -147,36 +148,34 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 			return err
 		}
 
-		var filename string       // path to Dockerfile
-		var origDockerfile string // used for error msg
+		filename := *dockerfileName // path to Dockerfile
 
 		if *dockerfileName == "" {
 			// No -f/--file was specified so use the default
-			origDockerfile = api.DefaultDockerfileName
-			*dockerfileName = origDockerfile
+			*dockerfileName = api.DefaultDockerfileName
 			filename = path.Join(absRoot, *dockerfileName)
-		} else {
-			origDockerfile = *dockerfileName
-			if filename, err = filepath.Abs(*dockerfileName); err != nil {
-				return err
-			}
+		}
 
-			// Verify that 'filename' is within the build context
-			if !strings.HasSuffix(absRoot, string(os.PathSeparator)) {
-				absRoot += string(os.PathSeparator)
-			}
-			if !strings.HasPrefix(filename, absRoot) {
-				return fmt.Errorf("The Dockerfile (%s) must be within the build context (%s)", *dockerfileName, root)
-			}
+		origDockerfile := *dockerfileName // used for error msg
 
-			// Now reset the dockerfileName to be relative to the build context
-			*dockerfileName = filename[len(absRoot):]
+		if filename, err = filepath.Abs(filename); err != nil {
+			return err
 		}
 
-		if _, err = os.Stat(filename); os.IsNotExist(err) {
-			return fmt.Errorf("Can not locate Dockerfile: %s", origDockerfile)
+		// Verify that 'filename' is within the build context
```

Comment:
I don’t think that should work since in that case Dockerfile is out of the context - even if its a symlink to something in the context.
-Doug

> On Feb 2, 2015, at 10:20 PM, Arnaud Porterie <notifications@github.com> wrote:
> 
> In api/client/commands.go <https://github.com/docker/docker/pull/10513#discussion_r23980251>:
> 
> >  		}
> >  
> > -		if _, err = os.Stat(filename); os.IsNotExist(err) {
> > -			return fmt.Errorf("Can not locate Dockerfile: %s", origDockerfile)
> > +		// Verify that 'filename' is within the build context
> So docker build -f Dockerfile /some/context invoked outside of /some/context will only work when Dockerfile is a symlink to a file rooted in /some/context. Let me try this.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/docker/docker/pull/10513/files#r23980251>.
>